setup.sh

```diff
@@ -50,6 +120,80 @@ if ! docker-compose version > /dev/null 2>&1; then
     fi
 fi
 
+# Check COMPOSE_PROFILES for existing configuration profile
+if get_profile "default"; then
+    PROFILE="default"
+elif get_profile "solar-only"; then
+    PROFILE="solar-only"
+else
+    # Prompt for configuration profile if not set
+    echo "Select configuration profile:"
+    echo ""
+    echo " 1 - default     (Powerwall w/ Gateway on LAN)"
+    echo " 2 - solar-only  (No Gateway - data retrieved from Tesla Cloud)"
+    echo ""
+    while :
+    do
+        read -r -p "Select profile: " response
```

Comment:
Minior thought: I wonder if we make the default 1 by also "Select profile [1]: " which means someone leaving it blank (enter) will get the default. We do that for other defaults.

Reply:
Also, the way the logic runs here, it will only prompt for configuration change (default or solar-only) the first time you run setup. I wonder if we should have a condition prompt like, "You are using the Powerwall (vs Solar-only) configuration, do you want to change that? [y/N]" to allow someone to switch. I suspect the switch would likely only be caused by an error running it first and selecting the wrong one, or if someone with Solar adds Powerwalls. What do you think?

Reply:
After a few more test and tweaking my idea, I'm leaning towards just leaving it as you have it. Keeping it simple for v3.0.0 makes the most sense. I'm going to test a bit more and then accept the PR if all looks well.

Reply:
As it happens, I had originally written the setup as you described, with a default on the "Select profile" prompt if the user pressed enter, and also allowing the chosen profile to be changed by re-running setup...

I later changed my thinking about this, and went with a forced prompt (user MUST make a selection, to prevent accidentally pressing enter and choosing the config profile without consciously looking at the options and making a selection). And also, NOT allowing the chosen profile to be changed later, as swapping and changing did not seem sensible (and could be achieved if you really wanted to by manually editing `compose.env`).

If we wanted to pivot to this method in future I had written some prelim methods to handle it as below:

```bash
# Read list of docker compose profiles from env file
. "${COMPOSE_ENV_FILE}"
IFS=',' read -a PROFILES <<< ${COMPOSE_PROFILES}
PROFILE="none"
current="1"
changed=0
index=-1
for i in "${!PROFILES[@]}"; do
    if [[ "${PROFILES[$i]}" == "default" ]]; then
        PROFILE="default"
        current="1"
        index=$i
        break
    elif [[ "${PROFILES[$i]}" == "solar-only" ]]; then
        PROFILE="solar-only"
        current="2"
        index=$i
        break
    fi
done

echo "Select Powerwall-Dashboard Setup Profile"
echo ""
echo "Current Setup Profile: ${PROFILE}"
echo ""
echo "1 - default     (Powerwall w/ Gateway on LAN)"
echo "2 - solar-only  (No Gateway - data retrieved from Tesla Cloud)"
echo ""
while [ 1 ]; do
    read -p "Select profile: [${current}] " response
    if [ -z "${response}" ]; then
        response="${current}"
    fi
    if [[ "${response}" == "1" ]]; then
        if [[ "${PROFILE}" != "default" ]]; then
            changed=1
        fi
        PROFILE="default"
    elif [[ "${response}" == "2" ]]; then
        if [[ "${PROFILE}" != "solar-only" ]]; then
            changed=1
        fi
        PROFILE="solar-only"
    else
        continue
    fi
    break
done

# Update list of docker compose profiles and save to env file
if [ $changed -eq 1 ]; then
    if [ $index -ge 0 ]; then
        PROFILES[$index]="${PROFILE}"
    else
        PROFILES+=("${PROFILE}")
    fi
    sed -i.bak -E "/(^#|^)COMPOSE_PROFILES=/d" "${COMPOSE_ENV_FILE}"
    echo "COMPOSE_PROFILES=$(IFS=,; echo "${PROFILES[*]}")" >> "${COMPOSE_ENV_FILE}"
fi
```

This was before the Docker Compose Helper Functions so would need some further changes if we were to use it. But that was the gist of what I had in mind originally.

However I do think the current implementation is best to keep it simple.

Reply:
On the default:

After I made that comment, in one of my test, I had hit enter twice after running the setup.  So your point is 100% valid. Let's keep it mandatory to enforce the user attention.

On future setup runs:

I'm fine with how it is for now.  However, the intent was that setup.sh could be run to fix a broken setup (user or other error).  For a future enhancement, it may be good to have setup.sh check compose.env and report what was selected and offer resetting.  Something like: "Existing configuration found: Your profile is set up for solar-only, do you wish to change that?"  If they do, nuke the COMPOSE_PROFILES setting and let setup.sh continue with setup (prompting for 1/2).